api-gateway/src/test/java/com/example/api/gateway/web/controller/GenerateControllerTest.java

```diff
@@ -0,0 +1,78 @@
+package com.example.api.gateway.web.controller;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.example.api.gateway.service.GenerationService;
+import com.example.api.gateway.service.exception.TemplateNotFoundException;
+
+@WebMvcTest(GenerateController.class)
```

Comment:
@coderabbitai, This module uses webflux so change everything in this class to use  `@WebFluxTest` and its framework